pkg/operator/ceph/disruption/clusterdisruption/add.go

```diff
@@ -108,13 +122,34 @@ func Add(mgr manager.Manager, context *controllerconfig.Context) error {
 	}
 
 	// Watch for main PodDisruptionBudget and enqueue the CephCluster in the namespace
+	if usePDBV1 {
+		err = c.Watch(
+			&source.Kind{Type: &policyv1.PodDisruptionBudget{}},
+			handler.EnqueueRequestsFromMapFunc(handler.MapFunc(func(obj client.Object) []reconcile.Request {
+				pdb, ok := obj.(*policyv1.PodDisruptionBudget)
+				if !ok {
+					// Not a pdb, returning empty
+					logger.Error("PDB handler received non-PDB")
+					return []reconcile.Request{}
+				}
+				namespace := pdb.GetNamespace()
+				req := reconcile.Request{NamespacedName: types.NamespacedName{Namespace: namespace}}
+				return []reconcile.Request{req}
+			}),
+			),
+			pdbPredicate,
+		)
+		if err != nil {
+			return err
+		}
+	}
 	err = c.Watch(
```

Comment:
Looks like we do need an `else` block here since there is no continue or return on the v1 condition.